app/route_rangers_api/views.py

```diff
@@ -137,19 +139,40 @@ def survey_p2(request, city: str, user_id: str = None):
     route_id = request.session.get("route_id")
 
     if request.method == "POST":
-        # check if route already exists
+
+        # post form data to database
         city_survey = CITIES_CHOICES_SURVEY[city]
         survey_answer = SurveyResponse(
             user_id_id=user_id, city=city_survey, route_id=route_id
         )
+        print(f"survey answer pg 2: {survey_answer}")
         update_survey = RiderSurvey2(request.POST, instance=survey_answer)
         # update and save
         update_survey.save()
 
+        # Get the line string data and update database
+        post_line_string = request.POST.get("lineString")
+        # Convert the string to a list of tuples
+        line_string_coords = json.loads(post_line_string)
+        # Create a LineString object
+        route = LineString(line_string_coords)
+        # Access the first and last points
+        starting_point = route.coords[0]
+        end_point = route.coords[-1]
+
+        # Update row in database
+        obj = SurveyResponse.objects.filter(user_id_id=user_id, route_id=route_id)
+
+        for item in obj:
```

Comment:
The combination of user_id and route_id should be unique so there is no need to do a for loop.
While it will not be inefficient, I think it can get confusing for someone reading our code.
I'll do this:
```
obj = SurveyResponse.objects.get(user_id_id=user_id, route_id=route_id)
obj.route = route
obj.starting_point = Point(starting_point)
obj.ending_point = Point(ending_point)
obj.save()
```